packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts

```diff
@@ -219,7 +215,7 @@ export class GutenbergEditorPage {
 		await frame.fill( selectors.blockSearch, blockName );
 		await frame.click( `${ selectors.blockInserterResultItem } span:text("${ blockName }")` );
 		// Confirm the block has been added to the editor body.
-		return await frame.waitForSelector( `*[aria-label="Block: ${ blockName }"].is-selected` );
+		return await frame.waitForSelector( `${ blockEditorSelector }.is-selected` );
```

Comment:
This assumption (that sidebar name matches aria label) has unfortunately turned out not to be true for every block! That's why I'm expanding this to have the in-editor selector be passed in explicitly.

Reply:
What is a block where the previous selector did not work?

Reply:
I haven't done a comprehensive pass, but there's definitely a handful I've already run across, like all the Embed variations, Blog List, and Contact Form.

I personally don't think it's too onerous on folks, as it's often just grabbing the `aria-label` for the block, which shouldn't take too long!

Alternatively, we could just ditch the validation piece of this method, and have just rely on auto-waiting for the next actions people take. This also means though that we're giving up returning the element handle, but I don't think that's too bad!

Or we could have two methods, one that does the validation and returns the handle, and one that doesn't? The latter could be used for folks who don't need to do a lot of work with the block.

Reply:
> like all the Embed variations, Blog List, and Contact Form.

Interesting, thanks for the examples. It would be nice to have a consistent identifier that matches what the text shows, but my suspicion is that even if we file a task for this, it would be low priority (if it can even be done).

> Or we could have two methods, one that does the validation and returns the handle, and one that doesn't?

What if the second parameter `blockEditorSelector` is made optional, and if entered the validation takes place?

Reply:
>No, that doesn't work. Hmm...I don't have a good solution, to be honest, without digging in myself. I can do that later today or Monday?

Sure thing!! You did a lot of the designing of the Editor POMs and patterns, and I definitely want what we go with here to fit the design you had envisioned, so let me know what you prefer! 👍

Reply:
After discussion in p1631305433179600-slack-CQD1HH4MA it was agreed there is no good solution that would a) not impose additional workload on test writer that is also b) reliable 100% with _every single block_.

Root of the problem is the inconsistency between block naming on the inserter as shown on the frontend and the `aria-label` attribute.